R/teal_data.R

```diff
@@ -32,25 +32,22 @@
 #'
 #' teal_data(x1, x2)
 teal_data <- function(...,
-                      join_keys,
+                      join_keys = teal.data::join_keys(),
                       code = "",
                       check = FALSE) {
   data_objects <- list(...)
-
-  is_cdisc <- vapply(
-    X = data_objects,
-    FUN.VALUE = logical(1),
-    FUN = function(x) {
-      inherits(x, "CDISCTealDataConnector") || inherits(x, "CDISCTealDatasetConnector") || inherits(x, "CDISCTealDataset")
-    }
+  checkmate::assert_list(
+    data_objects,
+    types = c("TealDataset", "TealDatasetConnector", "TealDataConnector")
   )
-
-  x <- if (any(is_cdisc)) {
-    CDISCTealData$new(..., check = check, join_keys = join_keys)
-  } else {
-    TealData$new(..., check = check, join_keys = join_keys)
+  if (inherits(join_keys, "JoinKeySet")) {
+    join_keys <- teal.data::join_keys(join_keys)
   }
 
+  get_primary_keys(data_objects, join_keys)
```

Comment:
We should probably rename as get implies we are returning the primary keys rather than updating join_keys or we could just return the list of join_key() and do:
 x <- get_primary_keys(data_objects, join_keys) 
 lapply(x, function(y) join_keys$mutate(y))